concepts/ranged-loops/introduction.md

```diff
@@ -0,0 +1,94 @@
+# Introduction
+
+When working with arrays or vectors in C++, iterating over their elements is a common task.  
+A range-based `for` loop provides a concise and readable way to access the elements without dealing with indices.
+
+## Basic Syntax
+
+A range-based `for` loop consists of the following parts:
+
+```cpp
+for (element_type temp_element : container) {
+    // Use temp_element inside the loop
+}
+```
+
+- `element_type`: The type of each element in the container.
+- `temp_element`: A temporary variable that holds the value of each element.
+- `container`: The array or vector to loop over.
+
+## Example Usage
+
```

Comment:
This took me a minute to work out but I like it! :)

Maybe it would be worth saying...

```suggestion
In this code, we loop over each word, retrieving the first letter each time, to reveal the obscured password.

```